### ISSUE TITLE
Clarify minimum Emacs version, drop obsolete support code

### ISSUE THREAD

**Issue by thblt** [alphapapa/outshine#40]
Outshine still contains support code for pre-24 Emacs.  We should set a minimum version, add it the documentation and drop useless compatibility code.

**Comment by thblt** [comment]
Obsolete code includes:

 - `outshine-compatible-face`
 - all the face definitions should be `:inherit outline-n` and nothing more.
 - what's the point of using `declare-function` instead of `eval-when-compile (require ...)`?
 - every reference to XEmacs.

**Comment by alphapapa** [comment]
Yes, I recommend targeting Emacs 25.1 as the earliest version.  Thanks.
